src/Microsoft.DotNet.Interactive.CSharp/AspNetKernelExtensions.cs

```diff
@@ -0,0 +1,169 @@
+    }
+}
+
+static LogLevelMonitor __AspNet_LogLevelMonitor = new LogLevelMonitor(new LoggerFilterOptions
+{
+    MinLevel = LogLevel.Warning,
+});
+
+static class Logging
+{
+    public static LogLevel MinLevel
+    {
+        set
+        {
+            __AspNet_LogLevelMonitor.CurrentValue = new LoggerFilterOptions { MinLevel = value };
+        }
+    }
+
+    public static bool EnableHttpClientTracing { get; set; }
+}
+
+class WriteLineDelegatingHandler : DelegatingHandler
+{
+    public WriteLineDelegatingHandler(HttpMessageHandler innerHandler)
+        : base(innerHandler)
+    {
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        if (Logging.EnableHttpClientTracing)
+        {
+            Console.WriteLine($""(HttpClient Request) {request}"");
+            if (request.Content != null)
+            {
+                Console.WriteLine(await request.Content.ReadAsStringAsync());
+            }
+        }
+
+        HttpResponseMessage response = await base.SendAsync(request, cancellationToken);
+
+        if (Logging.EnableHttpClientTracing)
+        {
+            Console.WriteLine($""(HttpClient Response) {response}"");
+            if (response.Content != null)
+            {
+                Console.WriteLine(await response.Content.ReadAsStringAsync());
+            }
+        }
+
+        return response;
+    }
+}
+
+private static int __AspNet_NextEndpointOrder;
+
+static IEndpointConventionBuilder MapAction(
+    this IEndpointRouteBuilder endpoints,
+    string pattern,
+    RequestDelegate requestDelegate)
+{
+    var order = __AspNet_NextEndpointOrder--;
+    var builder = endpoints.MapGet(pattern, requestDelegate);
+    builder.Add(b => ((RouteEndpointBuilder)b).Order = order);
+    return builder;
+}
+
+var HttpClient = new HttpClient(new WriteLineDelegatingHandler(new SocketsHttpHandler()));
+HttpClient.BaseAddress = new Uri(""http://localhost:5000/"");
+
+IApplicationBuilder App = null;
+IEndpointRouteBuilder Endpoints = null;
+
+var __AspNet_HostBuilder = Host.CreateDefaultBuilder()
+    .ConfigureServices(services =>
+    {
+        services.AddSingleton<IOptionsMonitor<LoggerFilterOptions>>(__AspNet_LogLevelMonitor);
+    })
+    .ConfigureWebHostDefaults(webBuilder =>
+    {
+        webBuilder.Configure(app => {
+            App = app.New();
+            App.UseRouting();
+            App.UseEndpoints(endpoints =>
+            {
+                Endpoints = endpoints;
+            });
+
+            app.Use(next =>
+                httpContext =>
+                   App.Build()(httpContext));
+        });
+    })
+    .ConfigureLogging(loggingBuilder =>
+    {
+        loggingBuilder.ClearProviders();
+        loggingBuilder.AddSimpleConsole(options => options.ColorBehavior = LoggerColorBehavior.Disabled);
+    });
+
+var __AspNet_HostRunAsyncTask = __AspNet_HostBuilder.Build().RunAsync();
```

Comment:
Is there a way to modify a host that is already running?

Reply:
Not yet, but I would love to add support for that.